src/Disks/BlobStorage/registerDiskBlobStorage.cpp

```diff
@@ -0,0 +1,173 @@
+#if !defined(ARCADIA_BUILD)
+#include <Common/config.h>
+#endif
+
+#include <Disks/DiskFactory.h>
+
+#if USE_AZURE_BLOB_STORAGE
+
+#include <Disks/BlobStorage/DiskBlobStorage.h>
+#include <Disks/DiskRestartProxy.h>
+#include <Disks/DiskCacheWrapper.h>
+#include <azure/identity/managed_identity_credential.hpp>
+#include <re2/re2.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+    extern const int PATH_ACCESS_DENIED;
+    extern const int FILE_DOESNT_EXIST;
+    extern const int FILE_ALREADY_EXISTS;
+}
+
+constexpr char test_file[] = "test.txt";
+constexpr char test_str[] = "test";
+constexpr size_t test_str_size = 4;
+
+
+void checkWriteAccess(IDisk & disk)
+{
+    auto file = disk.writeFile(test_file, DBMS_DEFAULT_BUFFER_SIZE, WriteMode::Rewrite);
+    file->write(test_str, test_str_size);
+}
+
+
+void checkReadAccess(IDisk & disk)
+{
+    auto file = disk.readFile(test_file);
+    String buf(test_str_size, '0');
+    file->readStrict(buf.data(), test_str_size);
+    if (buf != test_str)
+        throw Exception("No read access to disk", ErrorCodes::PATH_ACCESS_DENIED);
+}
+
+
+// TODO: make it a unit test ?
+void checkReadWithOffset(IDisk & disk)
+{
+    auto file = disk.readFile(test_file);
+    auto offset = 2;
+    auto test_size = test_str_size - offset;
+    String buf(test_size, '0');
+    file->seek(offset, 0);
+    file->readStrict(buf.data(), test_size);
+    if (buf != test_str + offset)
+        throw Exception("Failed to read file with offset", ErrorCodes::PATH_ACCESS_DENIED);
+}
+
+
+void checkRemoveAccess(IDisk & disk)
+{
+    // TODO: remove these checks if the names of blobs will be changed
+    if (!disk.checkUniqueId(test_file))
+        throw Exception(ErrorCodes::FILE_DOESNT_EXIST, "Expected the file to exist, but did not find it: {}", test_file);
+
+    disk.removeFile(test_file);
+
+    if (disk.checkUniqueId(test_file))
+        throw Exception(ErrorCodes::FILE_ALREADY_EXISTS, "Expected the file not to exist, but found it: {}", test_file);
+}
+
+
+void validate_endpoint_url(const String & endpoint_url)
+{
+    const auto * endpoint_url_pattern_str = R"(http(()|s)://[a-z0-9-]+\.blob\.core\.windows\.net/[a-z0-9-]+)";
```

Comment:
I think that at least for test purposes should be possible to connect to any url, for example to local docker image like https://hub.docker.com/_/microsoft-azure-blob-storage